isthmus/src/main/java/io/substrait/isthmus/expression/RexExpressionConverter.java

```diff
@@ -73,7 +92,115 @@ public Expression visitLiteral(RexLiteral literal) {
 
   @Override
   public Expression visitOver(RexOver over) {
-    throw new UnsupportedOperationException("RexOver not supported");
+    // maybe a aggregate function or a window function
+    var lowerBound = toWindowBound(over.getWindow().getLowerBound());
+    var upperBound = toWindowBound(over.getWindow().getUpperBound());
+    var sqlAggFunction = over.getAggOperator();
+    var argList = over.getOperands().stream().map(r -> ((RexSlot) r).getIndex()).toList();
+    boolean approximate = false;
+    int filterArg = -1;
+    var aggCall =
+        AggregateCall.create(
+            sqlAggFunction,
+            over.isDistinct(),
+            approximate,
+            over.ignoreNulls(),
+            argList,
+            filterArg,
+            null,
+            RelCollations.EMPTY,
+            over.getType(),
+            sqlAggFunction.getName());
+    var builder = Expression.Window.builder();
+    AbstractFunctionInvocation functionInvocation = null;
+    for (var converter : functionConverters) {
+      var funcInvocationOpt =
+          converter.convert(
+              inputRel,
+              inputType,
+              aggCall,
+              t -> {
+                RexNode r = (RexNode) t;
+                return r.accept(this);
+              });
+      if (funcInvocationOpt.isPresent()) {
+        functionInvocation = (AbstractFunctionInvocation) funcInvocationOpt.get();
+        if (functionInvocation instanceof AggregateFunctionInvocation) {
+          var aggFuncInvocation = (AggregateFunctionInvocation) functionInvocation;
+          var aggMeasure = Aggregate.Measure.builder().function(aggFuncInvocation).build();
+          builder.aggregateFunction(aggMeasure).hasNormalAggregateFunction(true);
+        } else if (functionInvocation instanceof WindowFunctionInvocation) {
+          var windowFuncInvocation = (WindowFunctionInvocation) functionInvocation;
+          var windowFunc =
+              ImmutableExpression.WindowFunction.builder().function(windowFuncInvocation).build();
+          builder.windowFunction(windowFunc).hasNormalAggregateFunction(false);
+        } else {
+          throw new RuntimeException(
+              String.format("Unexpected AbstractFunctionInvocation :%s", functionInvocation));
+        }
+      }
+    }
+    if (functionInvocation == null) {
```

Comment:
Maybe move this `if` block before line 126 for the case `functionInvocationOpt` is empty, for better code readability.

Reply:
Maybe that will not catch some failed cases.